setup.py

```diff
@@ -9,8 +9,7 @@
                  long_description=long_description,
                  long_description_content_type="text/markdown",
                  packages=setuptools.find_packages(exclude=['tests']),
-                 python_requires=('>=2.7, '
-                                  '!=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*, <3.13'),
+                 python_requires=('>=3.8, <3.13'),
```

Comment:
Upper bond on `3.13` is now replaced.

Reply:
Never, never ever put an upper bound on the Python version. It does not do what you thing it does and causes all sorts of breakage. Numba had to remove it, and it fixed the terrible error messages people were getting. NumPy briefly added an upper cap, but quickly backed off on it after it created a mess too. Pip back solves if this doesn't match - which is never the right thing to do for an upper cap! And if nothing is found, it produces a confusion "no package found" message rather than telling you that the package doesn't support 3.13. (But usually it finds a really old version of the package and tries to build that instead).

If you don't support the next Python version, you can put a cap in Conda - there it works, since conda can solve for the Python version. You can also put a "if sys.version_info >= (3, 13)` in your setup.py if you'd like.

Reply:
Numba added an upper bound in 0.52 and finally reverted it in 0.56 (they have always had the `<=MAX_VERSION` in `setup.py`). NumPy added it in 1.26.0 and reverted it in 1.26.2.

Reply:
Thanks @henryiii for the summary. I will revert back to my original removal of the Python version upper bound.